api/main.py

```diff
@@ -620,6 +622,69 @@ def get_latest_readings():
     return {"sensors": latest_readings}
 
 
+@app.post("/email")
+async def send_email(email: EmailSchema) -> JSONResponse:
+
+    msg = get_email_msg(
+        recipients=email.dict().get("recipient_list"), 
+        body=email.dict().get("body")
+        )
+
+    fm = get_fm()
+
+    await fm.send_message(msg)
+    return JSONResponse(
+        status_code=200, 
+        content={"message": "email has been sent"}
+        )
+
+
+@app.post("/send_alerts")
+async def send_alerts(email: AlertEmailSchema) -> JSONResponse:
+    alter_lvl = email.dict().get("alter_lvl")
+    sensors_latest_readings = get_latest_readings()['sensors']
+    recipient_list = email.dict().get("recipient_list")
+
+    # 1. Filter sensors with fill_level > a threshold
+    high_filled_sensors = [sensor for sensor in sensors_latest_readings if sensor.fill_level > alter_lvl]
+    if not high_filled_sensors:
+        email_schema = EmailSchema(
+            recipient_list=recipient_list, 
+            body="no sensors with fill level above " + str(alter_lvl)
+            )
+
+        response = await send_email(email_schema)
+        return response 
+
+    # 2. Format the data for the email
+    body = "Sensors with fill level above " + str(alter_lvl) + "%:\n\n"
+    for sensor in high_filled_sensors:
+        sensor_data = "\n".join([f"{key}: {value}" for key, value in sensor.dict().items()])
+        body += f"{sensor_data}\n\n"
+    
+    # 3. Send the email
+    email_schema = EmailSchema(
+        recipient_list=recipient_list, 
+        body=body
+    )
+    
+    response = await send_email(email_schema)
+    
+    return response
+
+# alert every 24 hours
+@app.on_event("startup")
+@repeat_every(seconds=60*60*24)
```

Comment:
Can you add a tech debt card to switch this to be sent at a specific time rather that timed with server startup? Not a big deal for now!

Reply:
done!  https://www.pivotaltracker.com/n/projects/2667313